packages/teams-js/src/public/appNotification.ts

```diff
@@ -0,0 +1,188 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to interact with the appNotification specific part of the SDK
+ */
+export namespace appNotification {
+  /**
+   * Data structure to represent appNotification information
+   */
+  export interface NotificationDisplayParam {
+    /**
+     * Notification title(maximum length: 75 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 1500 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    icon?: URL;
+    /**
+     * This would specify how long a notification would be displayed on the screen for (unit: seconds)
+     *
+     */
+    displayDurationInSeconds: number;
+    /**
+     * A url link to the page in which the notification would direct the user to.
+     */
+    notificationActionUrl: URL;
+  }
+
+  /**
+   * Data structure to represent appNotification information that would be sent to the host SDK
+   */
+  export interface NotificationDisplayParamForAppHost {
+    /**
+     * Notification title(maximum length: 75 characters)
+     */
+    title: string;
+    /**
+     * Notification content (maximum length: 1500 characters)
+     */
+    content: string;
+    /**
+     * This would represent an optional icon that can be displayed on the notification. It should have a max size of 49 pixels by 49 pixels
+     * If no icon is provided, the notification card would be displayed without an icon
+     * The url link to where the icon is stored should be provided as the input string
+     */
+    notificationIconAsSring?: string;
+    /**
+     * This would specify how long a notification would be displayed on the screen for (unit: seconds)
+     *
+     */
+    displayDurationInSeconds: number;
+    /**
+     * A url string type to the page in which the notification would direct the user to.
+     */
+    notificationActionUrlAsString: string;
+  }
+
+  /**
+   * This converts the notifcationActionUrl from a URL type to a string type for proper flow across the iframe
+   * @param notificationDisplayParam - appNotification display parameter with the notificationActionUrl as a URL type
+   * @returns a serialized object that can be sent to the host SDK
+   */
+  function serializeParam(notificationDisplayParam: NotificationDisplayParam): NotificationDisplayParamForAppHost {
```

Comment:
same for this. Mark it @internal and @hidden

Reply:
We don't need to generate documentation on the functions/interfaces that app developers don't need. 
Tagging @TrevorJoelHarris to double check this.